src/ratio.jl

```diff
@@ -0,0 +1,31 @@
+# Base's Rational type is too slow because of its penchant for calling gcd and div.
+# So we roll our own.
```

Comment:
I too would like to see a faster Rational type, though the overflow should have been fixed since #8672.